player/lua.c

```diff
@@ -1194,17 +1167,64 @@ static const struct fn_entry utils_fns[] = {
     FN_ENTRY(split_path),
     FN_ENTRY(join_path),
     FN_ENTRY(getpid),
-    FN_ENTRY(parse_json),
-    FN_ENTRY(format_json),
+    AF_ENTRY(parse_json),
+    AF_ENTRY(format_json),
     {0}
 };
 
+/* returns the talloc ctx which script_autofree_trampoline sets */
+static void *mp_lua_PITA(lua_State *L)
+{
+    void **ctx = (void**)lua_touserdata(L, lua_upvalueindex(1));
+    if (!ctx) {
+        lua_pushstring(L, "missing autofree ctx");
+        lua_error(L);
```

Comment:
I think just an assert() would be better. This is a logic error that can only happen if the C code does something wrong.

Reply:
Correct. And even the assert can be removed once we change the prototype to have a ctx argument, so it's enforced at compile time. Probably.